strawberry/cli/commands/codegen.py

```diff
@@ -78,8 +84,26 @@ def _load_plugin(plugin_path: str) -> Type[QueryCodegenPlugin]:
     return plugin
 
 
-def _load_plugins(plugins: List[str]) -> List[QueryCodegenPlugin]:
-    return [_load_plugin(plugin)() for plugin in plugins]
+def _load_plugins(plugin_ids: List[str], query: Path) -> List[QueryCodegenPlugin]:
+    plugins = []
+    for ptype_id in plugin_ids:
+        ptype = _load_plugin(ptype_id)
+        # inject arguments into the plugin's `__init__` based on
+        # the names in the signature (e.g., similar to flake8 plugins).
+        sig = inspect.signature(ptype)
+        kwargs = {}
+        if "query" in sig.parameters:
+            kwargs["query"] = query
+        if "query_file" in sig.parameters:
+            kwargs["query_file"] = query
+
+        plugin = ptype(**kwargs)
+        if not hasattr(plugin, "query"):
+            plugin.query = query
```

Comment:
is this to keep backwards compatibility?

Reply:
yeah, I did a lot of shenanigans to try to preserve compatibility with any existing plugins that might be in the wild.

If we don't care about that, then we can probably simplify this a good bit.

Reply:
I think it might be fine to break compatibility for this, we can just add a note in our breaking changes files. We could do a codemod too but it is probably not worth doing 😊